### ISSUE TITLE
getDuration() unpredictably returns either Infinity and NaN when used in the "ready" event listener

### ISSUE THREAD

**Issue by Johnrobmiller** [katspaugh/wavesurfer.js#3104]
In this code:

```
wavesurfer.current.once('ready', () => {
  setReady(true)
  handleSetDuration()
})
```

I am wanting to know how long an audio file is after it been processed and is ready to play. However, there seems to be a race condition where the duration will initially be either NaN or Infinity until a brief period has passed. The workaround I am using for this is to poll the method every 50ms until it gives a real number:

```
  const handleSetDuration = async () => {
    const checkDuration = (duration: number | undefined | null) => {
      if (!duration) return
      if (duration === Infinity) return
      if (isNaN(duration)) return
      return duration
    }

    // Waits one second then gives up
    for (let i = 0; i < 40; i++) {
      const newDuration = checkDuration(wavesurfer?.current?.getDuration())
      if (!newDuration) {
        await new Promise((resolve) => setTimeout(resolve, 50))
        continue
      }

      setDuration(newDuration)
      break
    }
  }
```

**Comment by katspaugh** [comment]
How do you load the audio? With peaks or without? What type of file is it?

**Comment by mahdimirmohamadi** [comment]
i was facing the same problem just yesterday. i do not fully understand your code but it seems to me that you can do it much more simpler like this(this is the way i used):

`
// Load audio from a remote url.
      waveformRef.current.load(url);

   waveformRef.current.on('ready', () => {
        // @ts-ignore
        const duration = waveformRef.current.getDuration();
        setAudioDuration(duration);
      });
`

in summary, just save the Duration in a variable and then set it as your state. i think problem is kinda about the closures and the fact that state updates in React are asynchronous(not sure though!)

**Comment by katspaugh** [comment]
Might be fixed by #3106.

**Comment by katspaugh** [comment]
I'll go ahead and close this issue, please feel free to reopen it if it's still not fixed.
